ReactSkia/sdk/NotificationCenter.h

```diff
@@ -0,0 +1,153 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell 
+// copies of the Software, and to permit persons to whom the Software is 
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in 
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE 
+// SOFTWARE.
+//
+/*
+* Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+*
+* Use of this source code is governed by a BSD-style license that can be
+* found in the LICENSE file.
+*/
+
+#include <iostream>
+#include <functional>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <list>
+#include <algorithm>
+
+#pragma once
+
+class NotificationCenter
+{
+    private:
+        struct ListenerBase
+        {
+            ListenerBase() {}
+
+            ListenerBase(unsigned int i)
+            : id(i) {}
+
+            virtual ~ListenerBase() {}
+
+            unsigned int id;
+        };
+
+        template <typename... Args>
+        struct Listener : public ListenerBase
+        {
+            Listener() {}
+
+            Listener(unsigned int i, std::function<void (Args...)> c)
+            : ListenerBase(i), cb(c) {}
+
+            std::function<void (Args...)> cb;
+        };
+
+        static std::mutex mutex;
+        static unsigned int last_listener;
+        static std::multimap<std::string, std::shared_ptr<ListenerBase>> listeners;
+
+        NotificationCenter(const NotificationCenter&) = delete;  
+        const NotificationCenter& operator = (const NotificationCenter&) = delete;
+    public:
+        NotificationCenter() {
+            NotificationCenter::last_listener = 0;
+        }
+
+        ~NotificationCenter() {}
+
+        template <typename... Args>
+        unsigned int addListener(std::string eventName, std::function<void (Args...)> cb);
+
+        //unsigned int addListener(std::string eventName, std::function<void ()> cb);
+
+        template <typename... Args>
+        unsigned int on(std::string eventName, std::function<void (Args...)> cb);
+        
+        //unsigned int on(std::string eventName, std::function<void ()> cb);
+        void removeListener(unsigned int listener_id);
+
+        template <typename... Args>
+        void emit(std::string eventName, Args... args);
+
+};
+
+template <typename... Args>
+unsigned int NotificationCenter::addListener(std::string eventName, std::function<void (Args...)> cb)
+{
+    if (!cb)
+    {
```

Comment:
I assume this file is taken from notificatio center and modified. Should we follow same coding style as RNS ? i.e 
if () {
}

Reply:
done